docs/demultiplexing.md

```diff
@@ -0,0 +1,171 @@
+        }},
+        {"config": {
+            "name": "Model_B",
+            "base_path": "/models/model_b"
+        }},
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "my_pipeline_with_demultiplexer",
+            "inputs": ["pipeline_input_name"],
+            "nodes": [
+                {
+                    "name": "Model_A_node",
+                    "model_name": "Model_A",
+                    "type": "DL model",
+                    "demultiply_count": 3,  <----------------- this parameter specifies how many branches should be spawned
+                    "inputs": [
+                        {"input": {"node_name": "request",
+                                   "data_item": "pipeline_input_name"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output_A",
+                         "alias": "output_A"},
+                        {"data_item": "output_B",
+                         "alias": "output_B"}
+                    ]
+                },
+                {
+                    "name": "Model_B_node",
+                    "model_name": "Model_B",
+                    "type": "DL model",
+                    "inputs": [
+                        {"input_A": {"node_name": "Model_A_node",
+                                     "data_item": "output_A"}},
+                        {"input_B": {"node_name": "Model_A_node",
+                                     "data_item": "output_B"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                },
+                {
+                    "name": "Model_C_node",
+                    "model_name": "Model_C",
+                    "type": "DL model",
+                    "gather_from_node": "Model_A_node",  <----------------- this parameter specifies which node to gather from (must be demultiplexer and in LIFO order)
+                    "inputs": [
+                        {"input": {"node_name": "Model_B_node",
+                                     "data_item": "output"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"pipeline_output_name": {"node_name": "Model_C_node",
+                                          "data_item": "output"}}
+            ]
+        }
+    ]
+}
+```
+
+## Pipeline configuration restrictions
+There are several restricted configurations in regards to demultiplexing and gathering:
+- You can gather only from nodes with `demultiply_count` specified (demultiplexer nodes)
+- Demultiplexer nodes and gathering nodes should be in LIFO order. Meaning you need to gather nodes starting from closest demultiplexer going upstream in Directed Acyclic Graph Scheduler
```

Comment:
```suggestion
- Demultiplexer nodes and gathering nodes should be in LIFO order. Meaning you need to gather nodes starting from closest demultiplexer going upstream in defined directed acyclic graph.
```
?

Reply:
All around the docs we capitalize first letters of DAG. Should we change it everywhere? @dtrawins

Reply:
Those capitals here looks to me that you are referring to the feature itself. In this case you are referring to one pipeline that you are defining - that's why in this context it looks odd especially with "Scheduler".

Reply:
Done